nautobot/utilities/testing/filters.py

```diff
@@ -0,0 +1,40 @@
+from django.test import tag
+from nautobot.utilities.testing.views import TestCase
+
+
+@tag("unit")
+class FilterTestCases:
+    class FilterTestCase(TestCase):
+        """Base class for testing of FilterSets."""
+
+        queryset = None
+        filterset = None
+
+        def test_id(self):
+            """Verify that the filterset supports filtering by id."""
+            params = {"id": self.queryset.values_list("pk", flat=True)[:2]}
+            filterset = self.filterset(params, self.queryset)
+            self.assertTrue(filterset.is_valid())
+            self.assertEqual(filterset.qs.count(), 2)
+
+        def test_invalid_filter(self):
+            """Verify that the filterset reports as invalid when initialized with an unsupported filter parameter."""
+            params = {"ice_cream_flavor": ["chocolate"]}
```

Comment:
Not saying it's likely a filter set will contain an `ice_cream_flavor` filter, but could we make this obscenely long to make it really hard to have a conflict? Maybe even just a random string, say `xf7qmnb1ec`?

Reply:
IMHO this is already sufficiently long and safe.